modules/backendscheduler/provider/compaction.go

```diff
@@ -344,29 +347,19 @@ func (p *CompactionProvider) newBlockSelector(tenantID string) (blockselector.Co
 		blocklist     = make([]*backend.BlockMeta, 0, len(fullBlocklist))
 	)
 
-	// Query the work for the jobs and build up a list of UUIDs which we can match against.
+	// Query the work for the jobs and build up a list of UUIDs which we can match against and skip on the selector.
 	var (
 		perTenantBlockIDs = make(map[backend.UUID]struct{})
 		bid               backend.UUID
 		err               error
-
-		jobTenant string
-		jobType   tempopb.JobType
-		jobStatus tempopb.JobStatus
 	)
 
 	for _, job := range p.sched.ListJobs() {
-		jobTenant = job.Tenant()
-		if jobTenant != tenantID {
-			continue
-		}
-
-		if jobType = job.GetType(); jobType == tempopb.JobType_JOB_TYPE_UNSPECIFIED {
+		if job.Tenant() != tenantID {
 			continue
 		}
 
-		jobStatus = job.GetStatus()
-		if jobStatus != tempopb.JobStatus_JOB_STATUS_RUNNING && jobStatus != tempopb.JobStatus_JOB_STATUS_UNSPECIFIED {
+		if job.GetType() == tempopb.JobType_JOB_TYPE_UNSPECIFIED {
```

Comment:
When can a job have unspecified type?

Reply:
It shouldn't ever happen, but its in the proto, so this is a protection.

Reply:
I think technically we could drop this here, since the check is performed in the GetCompactionInput method.